documentation/prometheus-mixin/alerts.libsonnet

```
@@ -0,0 +1,192 @@
+            alert: 'PrometheusTSDBWALCorruptions',
+            expr: |||
+              tsdb_wal_corruptions_total{%(prometheusSelector)s} > 0
+            ||| % $._config,
+            'for': '4h',
+            labels: {
+              severity: 'warning',
+            },
+            annotations: {
+              message: '{{$labels.job}} at {{$labels.instance}} has a corrupted write-ahead log (WAL).',
+            },
+          },
+          {
+            alert: 'PrometheusNotIngestingSamples',
+            expr: |||
+              rate(prometheus_tsdb_head_samples_appended_total{%(prometheusSelector)s}[5m]) <= 0
+            ||| % $._config,
+            'for': '10m',
+            labels: {
+              severity: 'warning',
+            },
+            annotations: {
+              message: "Prometheus {{ $labels.namespace }}/{{ $labels.pod}} isn't ingesting samples.",
+            },
+          },
+          {
+            alert: 'PrometheusTargetScrapesDuplicate',
+            expr: |||
+              increase(prometheus_target_scrapes_sample_duplicate_timestamp_total{%(prometheusSelector)s}[5m]) > 0
+            ||| % $._config,
+            'for': '10m',
+            labels: {
+              severity: 'warning',
+            },
+            annotations: {
+              message: '{{$labels.namespace}}/{{$labels.pod}} has many samples rejected due to duplicate timestamps but different values',
+            },
+          },
+          {
+            alert: 'PrometheusRemoteStorageFailures',
+            expr: |||
+              (rate(prometheus_remote_storage_failed_samples_total{%(prometheusSelector)s}[1m]) * 100)
+                /
+              (rate(prometheus_remote_storage_failed_samples_total{%(prometheusSelector)s}[1m]) + rate(prometheus_remote_storage_succeeded_samples_total{%(prometheusSelector)s}[1m]))
+                > 1
+            ||| % $._config,
+            'for': '15m',
+            labels: {
+              severity: 'critical',
+            },
+            annotations: {
+              message: 'Prometheus failed to send {{ printf "%.1f" $value }}% samples',
+            },
+          },
+          {
+            alert: 'PrometheusRemoteWriteBehind',
+            expr: |||
+              prometheus_remote_storage_highest_timestamp_in_seconds{%(prometheusSelector)s}
+                - on(job, instance) group_right
+              prometheus_remote_storage_queue_highest_sent_timestamp_seconds{%(prometheusSelector)s}
+                > 120
+            ||| % $._config,
+            'for': '15m',
+            labels: {
+              severity: 'critical',
+            },
+            annotations: {
+              message: 'Prometheus remote write is {{ printf "%.1f" $value }}s behind.',
+            },
+          },
+          {
+            alert: 'PrometheusRuleFailures',
+            'for': '15m',
+            expr: |||
+              rate(prometheus_rule_evaluation_failures_total{%(prometheusSelector)s}[1m]) > 0
```

```suggestion
              rate(prometheus_rule_evaluation_failures_total{%(prometheusSelector)s}[2m]) > 0
```
or make this range configurable because it needs to be larger than the rule eval interval

---

I think 5m is a safe bet. And it doesn't really has to be shorter if we only alert after 15m anyway.